tests/Builder/SourceFileRendererTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Tests\Builder;
+
+use DateTimeImmutable;
+use Doctrine\Website\Builder\SourceFile;
+use Doctrine\Website\Builder\SourceFileParameters;
+use Doctrine\Website\Builder\SourceFileRenderer;
+use Doctrine\Website\Controller\ControllerExecutor;
+use Doctrine\Website\Site;
+use Doctrine\Website\Twig\TwigRenderer;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class SourceFileRendererTest extends TestCase
+{
+    /** @var ControllerExecutor|MockObject */
+    private $controllerExecutor;
+
+    /** @var TwigRenderer|MockObject */
+    private $twigRenderer;
+
+    /** @var Site|MockObject */
+    private $site;
+
+    /** @var SourceFileRenderer */
+    private $sourceFileRenderer;
+
+    public function testRenderWithContentBlock() : void
+    {
+        $date                 = new DateTimeImmutable('2018-09-01');
+        $sourceFile           = $this->createMock(SourceFile::class);
+        $sourceFileParameters = new SourceFileParameters(['controller' => 'TestController']);
+        $controllerResult     = ['test' => true];
+        $contents             = 'Test';
+
+        $sourceFile->expects(self::once())
+            ->method('getDate')
+            ->willReturn($date);
+
+        $sourceFile->expects(self::once())
+            ->method('getParameters')
+            ->willReturn($sourceFileParameters);
+
+        $sourceFile->expects(self::once())
+            ->method('isLayoutNeeded')
+            ->willReturn(true);
+
+        $sourceFile->expects(self::once())
+            ->method('getParameter')
+            ->with('layout')
+            ->willReturn('default');
+
+        $this->controllerExecutor->expects(self::once())
+            ->method('execute')
+            ->with($sourceFile)
+            ->willReturn($controllerResult);
+
+        $this->twigRenderer->expects(self::once())
+            ->method('render')
+            ->with('{% extends "layouts/default.html.twig" %}{% block content %}Test{% endblock %}', [
+                'page' => [
+                    'date' => $date,
+                    'controller' => 'TestController',
+                ],
+                'site' => $this->site,
+                'test' => true,
+            ]);
+
+        $this->sourceFileRenderer->render(
+            $sourceFile,
+            $contents
+        );
+    }
+
+    public function testRenderWithoutContentBlock() : void
+    {
+        $date                 = new DateTimeImmutable('2018-09-01');
+        $sourceFile           = $this->createMock(SourceFile::class);
+        $sourceFileParameters = new SourceFileParameters(['controller' => 'TestController']);
+        $controllerResult     = ['test' => true];
+        $contents             = '{% block content %}Testing{% endblock %}';
```

Comment:
And with `$content`, this name would fit to the one of the test method. It's easier to read.